### DISCUSSION TITLE
unwrap may not return previous value

### DISCUSSION THREAD

**Discussion by satoren** [pmndrs/jotai#2214] [Bug report] [answered]
I implemented debounce with unwrap thinking it would be easy to implement, but often forgot the previous value.
https://codesandbox.io/s/react-typescript-forked-vjc9g4?file=/src/debounceAtom.ts

I'm ready to create a PR for the fix.
https://github.com/pmndrs/jotai/commit/04a880ba612aa61017d62fe5d815e68712573881

**Comment by dai-shi** [comment, marked as answer]
Oh, nice catch. Although, your fix seems a bit weird, please open a PR so that I can try it out.

**Comment by satoren** [comment]
Fixed at https://github.com/pmndrs/jotai/pull/2216, v2.5.1
